### ISSUE TITLE
Change of scope and potential namespace update

### ISSUE THREAD

**Issue by SimonDarksideJ** [Unity-UI-Extensions/com.unity.uiextensions#138]
<sup>Issue created by Rahul Raman as [Bitbucket Issue #&#8203;102](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/102) on 2017.01.01 09:36.</sup>
I noticed, with the inclusion of the custom serialization  and CLFZ2 scripts, that this repository could (and has begun to) extend beyond merely extensions for Unity's UI system and into extensions for Unity as a whole.

Perhaps the scope of the project must be revised at this point.

Toward that direction, I propose that the common namespace for scripts be updated to
UnityEngine.Extensions.xyz, where "xyz" is replaced by the body of features your script targets.

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.01.01 11:51](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/102#comment-34323028), s commented:</sup>
As stated in the update where I added them, there were a special case only, I' m not a big fan of creating "uber" libraries as they get far too bloated far too easily.

That is why the "Utilities" folder is so light, else I'd create a separate UIExtensions utilities project.

If you're happy with that, can we close this issue?

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.01.04 11:55](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/102#comment-34323026), Rahul Raman commented:</sup>
I understand where you're coming from and I agree that an uber-library extending all feature sets of the engine can suffer from bloat very quickly.

But I can see the merits of having an organized, well-maintained collection of scripts that improve all areas of the engine out-weighing the trouble that comes from keeping the project focused.

The alternative might be creating dedicated projects for each set of scripts targeting a specific feature of the engine, like you've already mentioned (which, however, comes with the trouble of mapping dependencies).

Even so, is neither approach worth considering? :)

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.01.04 12:20](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/102#comment-34323027), @SimonDarksideJ commented:</sup>
The scripts are currently well organized in separate folders under the main Extensions namespace.
A few have sub namespaces where it adds value in doing so.

But adding more namespaces wouldn't improve the codability through Unity (in fact namespaces have no effect on Unity it'self). With Unity the only reason to have a namespace, is to ensure it's cross-compiler does not get confused.

Granted in proper programming, more is better, but this is Unity :P

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.01.04 13:06](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/102#comment-34323025), Rahul Raman commented:</sup>
Haha
No, no. I wasn't talking about organization via namespaces at all. I was just speaking in favor of the uber-library project :P

(I've edited my previous comment to hopefully make that evident.)

What I'm saying is:
I know you're not in favor of the idea, and I recognize that it may bring with it a number of problems, but isn't it worth pursuing, anyway? Scripts like the CLFZ2 compression one mentioned previously will find a "place where it is home".

If you think it's not a suggestion that must be dismissed immediately, maybe we could create a poll and see what other users/contributors think?

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.01.04 14:22](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/102#comment-34323024), @SimonDarksideJ commented:</sup>
Still struggling to see the direction you're proposing.

So what would the poll be asking?
* Should this stay as a UI project
* Or create an uber asset project.

Personally, if you want an uber project, then it might be better to create a new Project / Library and had the UI Extensions as a subproject of that one.

The primary goal with the UI Extensions project is the Unity UI and extended it.

The Compression and Serialisation parts were only added to help towards that since a lot of people use data binding frameworks (like Unity Foundation) and needs to save / load data. (and I thought the scripts were cool and hard to get hold of.  If anyting else, if an uber project was created and UI Extensions was a part of that, then I'd happily strip out those parts and move them to the uber project rather than start expanding this one beyond UI.

Thoughts?

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.01.04 14:40](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/102#comment-34323023), Rahul Raman commented:</sup>
That's a completely valid point. And yes, turning this into an uber-library
is exactly what I was proposing -
one where scripts (and not just UI ones) that were cool and hard to get to
get a hold of could go to.

The reason I proposed the idea of expanding the scope of this project to
accomodate extensions to all portions of Unity instead of just creating a
new repository/project for the same was because this one has already begun
to receive a push from the developer community and I figured it'd,
therefore, gain momentum much faster.

Just an idea that I thought might be interesting/worth discussing.

However, if that conflicts with the goal of this particular repository
and/or lead to lower productivity, then I'd rather not change a thing :)
